### PR TITLE
Add read level consistency of unstrict majority and error metadata

### PR DESCRIPTION
**Pull request by robskillington**
cc @xichen2020 @martin-mao @cw9 @kobolog 

This introduces the concept of "unstrict majority" for reads such that applications can use to always attempt to achieve majority but in the case when all nodes have answered and less than the majority of nodes have responded successfully to go ahead nonetheless and return with a successful result.  

This means that if two nodes are healthy the application will always read from the majority of nodes to ensure data is merged from both healthy nodes to read what you put in at consistency level majority.  It does mean in an extreme failure case of two nodes being unhealthy that the application can continue to serve reads knowingly with no guarentees of serving partial reads.

### CODE REVIEW

File: client/options.go [hunk 1]
```diff
@@ -178,6 +182,7 @@ func newOptions() *options {
 		clockOpts:                             clock.NewOptions(),
 		instrumentOpts:                        instrument.NewOptions(),
 		consistencyLevel:                      defaultConsistencyLevel,
```
Comment (xichen2020) [inline client/options.go]: Should we call this `writeConsistencyLevel` assuming it's used for write consistency purposes?

File: client/types.go [hunk 2]
```diff
@@ -59,6 +59,57 @@ const (
...
+	case ConnectConsistencyLevelNone:
+		return "ConnectConsistencyLevelNone"
+	case ConnectConsistencyLevelOne:
+		return "ConnectConsistencyLevelOne"
+	case ConnectConsistencyLevelMajority:
+		return "ConnectConsistencyLevelMajority"
+	case ConnectConsistencyLevelAll:
+		return "ConnectConsistencyLevelAll"
+	}
+	return ""
```
Comment (xichen2020) [inline client/types.go]: Might be better to return "ConnectConsistencyLevelUnknown" so it's easier to debug?

File: client/types.go [hunk 3]
```diff
@@ -59,6 +59,57 @@ const (
...
+	case ReadConsistencyLevelOne:
+		return "ReadConsistencyLevelOne"
+	case ReadConsistencyLevelUnstrictMajority:
+		return "ReadConsistencyLevelUnstrictMajority"
+	case ReadConsistencyLevelMajority:
+		return "ReadConsistencyLevelMajority"
+	case ReadConsistencyLevelAll:
+		return "ReadConsistencyLevelAll"
+	}
+	return ""
```
Comment (xichen2020) [inline client/types.go]: Might be better to return "ReadConsistencyLevelUnknown" so it's easier to debug?

**Comment by kobolog** [comment]
As a sidenote, I suggest that we start filling in PR description & details apart from CC'ing people to review it. While we can just talk to each other to figure out what's this about, I'd expect new team members and potential external users to be at least confused, most likely end up ignoring PRs and even feel excluded.

File: client/errors.go [hunk 4]
```diff
@@ -48,3 +50,96 @@ func IsBadRequestError(err error) bool {
...
+}
+
+func (e consistencyResultErr) Error() string {
+	return fmt.Sprintf(
+		"failed to meet %s with %d/%d success, %d nodes responded, errors: %v",
+		e.level.String(), e.success, e.enqueued, e.responded, e.errs)
+}
+
+func (e consistencyResultErr) numResponded() int {
+	return int(e.responded)
```
Comment (robskillington) [inline client/errors.go]: Ah yeah, these used to be int32's.  Ta

File: client/types.go (hunk 2 above)
Comment (robskillington) [inline client/types.go]: Yeah, sounds good.

**Comment by robskillington** [comment]
Added a PR description.  Agreed @kobolog.

**Pull request merged**